wrapper/src/main/java/software/amazon/jdbc/plugin/failover2/FailoverConnectionPlugin.java

```diff
@@ -492,7 +494,7 @@ protected void failoverWriter() throws SQLException {
         throw new FailoverFailedSQLException(Messages.get("Failover.unableToConnectToWriter"));
       }
 
-      final List<HostSpec> updatedHosts = this.pluginService.getHosts();
+      final List<HostSpec> updatedHosts = this.pluginService.getAllHosts();
```

Comment:
Does it make sense to use `getHosts()` here?

Reply:
I think it makes sense to use all hosts, the only thing updatedHosts is used for is to identify the writer, so it makes sense to include all hosts in case the subset of hosts doesn't include the writer. Below this we check if the identified writer is in the list of allowed hosts and throw an exception if it isn't